components/Carousel/Carousel.js

```diff
@@ -1,5 +1,45 @@
 class Carousel {
+	constructor(mainElement){
+		// assign parent/main element
```

Comment:
Great use of comments, it really lets others know what you are doing. Keep this up.